src/spec.rs

```diff
@@ -0,0 +1,102 @@
+//! Helper functions defined in the Zcash Protocol Specification.
+
+use std::iter;
+
+use blake2b_simd::{Hash, Params};
+use ff::PrimeField;
+use group::Curve;
+use halo2::{
+    arithmetic::{CurveAffine, CurveExt, FieldExt},
+    pasta::pallas,
+};
+
+use crate::{constants::L_ORCHARD_SCALAR, primitives::sinsemilla};
+
+const PRF_EXPAND_PERSONALIZATION: &[u8; 16] = b"Zcash_ExpandSeed";
+
+/// $\mathsf{ToBase}^\mathsf{Orchard}(x) := LEOS2IP_{\ell_\mathsf{PRFexpand}}(x) (mod q_P)$
+///
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][§4.2.3].
+///
+/// [§4.2.3]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn to_base(hash: Hash) -> pallas::Base {
+    pallas::Base::from_bytes_wide(hash.as_array())
+}
+
+/// $\mathsf{ToScalar}^\mathsf{Orchard}(x) := LEOS2IP_{\ell_\mathsf{PRFexpand}}(x) (mod r_P)$
+///
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][§4.2.3].
+///
+/// [§4.2.3]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn to_scalar(hash: Hash) -> pallas::Scalar {
```

Comment:
Strictly speaking, `ToScalar` takes an array of bytes as input, not just a Blake2b `Hash`. If it's being defined, shall we make it accurate, especially since that `Hash` is turning into `[u8; 64]` anyway? Same with `ToBase`